demos/tests/run_tests.py

```diff
@@ -133,23 +154,33 @@ def option_to_args(key, value):
 
             print('Fixed arguments:', ' '.join(map(shlex.quote, fixed_args)))
             print()
-
-            for test_case_index, test_case in enumerate(demo.test_cases):
+            case_index = 0
+            device_args = demo.device_args(args.devices.split(', '))
+            for test_case in demo.test_cases:
                 case_args = [demo_arg
                     for key, value in sorted(test_case.options.items())
                     for demo_arg in option_to_args(key, value)]
 
-                print('Test case #{}:'.format(test_case_index + 1),
-                    ' '.join(shlex.quote(str(arg)) for arg in case_args))
-                print(flush=True)
-
-                try:
-                    subprocess.check_output(fixed_args + case_args,
-                        stderr=subprocess.STDOUT, universal_newlines=True)
-                except subprocess.CalledProcessError as e:
-                    print(e.output)
-                    print('Exit code:', e.returncode)
-                    num_failures += 1
+                pipeline = [option_to_args(key, value)[1] for key, value in test_case.options.items() if "-m" in key]
+
+                for device, dev_arg in device_args.items():
+                    case_index +=1
```

Comment:
I don't think we should increment the index for each device, since then there's a lack of correspondence between output and the elements of `demo.test_cases`. It would be better to print the device as part of the case ID, e.g. "Test case #5/GPU:".